### PR TITLE
Add whitelist and blacklist restrictions on attacking entity types

### PR DESCRIPTION
**Pull request by drunderscore**
Similar to the Block Type Restriction, this Entity Type Restriction will prevent attacking entities if they are on the blacklist/not on the
whitelist.

This also introduces a helper restriction class, `EntityRestriction`.

I've put `minecraft:villager` as a default, as attacking them will increase trade prices, so you often don't want to attack them.

### CODE REVIEW

File: src/main/java/fi/dy/masa/tweakeroo/tweaks/MiscTweaks.java [hunk 1]
```diff
@@ -222,6 +228,28 @@ private static void doPotionWarnings(PlayerEntity player)
         }
     }
 
+    public static boolean isEntityAllowedByAttackingRestriction(EntityType<?> type)
+    {
+        if(MiscTweaks.ENTITY_TYPE_ATTACK_RESTRICTION.isAllowed(type) == false)
```
Comment (maruohon) [inline src/main/java/fi/dy/masa/tweakeroo/tweaks/MiscTweaks.java]: I always use a space in the if/for/while before the opening parenthesis. (Same thing in the Mixin.)

File: src/main/java/fi/dy/masa/tweakeroo/util/EntityRestriction.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.List;
+import java.util.Set;
+
+public class EntityRestriction extends UsageRestriction<EntityType<?>>
+{
+    @Override
+    protected void setValuesForList(Set<EntityType<?>> set, List<String> names)
+    {
+        for (var name : names)
```
Comment (maruohon) [inline src/main/java/fi/dy/masa/tweakeroo/util/EntityRestriction.java]: I think I would write this as:
```java
            for (String name : names)
            {
                try
                {
                    Optional<EntityType<?>> type = Registry.ENTITY_TYPE.getOrEmpty(new Identifier(name));

                    if (type.isPresent())
                    {
                        set.add(type.get());
                        continue;
                    }
                }
                catch (Exception ignore) {}

                Tweakeroo.logger.warn("Invalid entity name in a black- or whitelist: '{}'", name);
            }
```

(Style notes: I don't like `var` to "hide" the type, and I use curly braces even for single line blocks.)

**Review by maruohon — CHANGES_REQUESTED** [review]
Thanks. I left a couple of small code style notes, and one actual code change request.

**Comment by drunderscore** [comment]
I've implemented all the suggested changes.
Apologies if the force-push isn't appreciated here, I'm used to doing it to keep merges clean of "fixup" type commits.

**Review by maruohon — APPROVED** [review]

**Pull request merged**